app/controllers/dogs_controller.rb

```diff
@@ -0,0 +1,57 @@
+class DogsController < ApplicationController
+  def new
+    @user = User.find(params[:id])
+    @dog = Dog.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @dog = Dog.new(dog_params)
+    @dog.user = current_user
+    if @dog.save
+      flash[:success] = 'New Dog Added!'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.